### PR TITLE
Use platform appropriate filepaths

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Works on my computer?

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_system/targets/assets.dart [hunk 1]
```diff
@@ -34,7 +34,7 @@ Future<Depfile> copyAssets(Environment environment, Directory outputDirectory) a
     assetBundle.entries.entries.map<Future<void>>((MapEntry<String, DevFSContent> entry) async {
       final PoolResource resource = await pool.request();
       try {
-        final File file = fs.file(fs.path.join(outputDirectory.path, entry.key));
+        final File file = fs.file(outputDirectory.uri.resolve(entry.key));
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/build_system/targets/assets.dart]: Is this fix needed anywhere else?

File: packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart [hunk 2]
```diff
@@ -47,14 +45,13 @@ flutter:
...
-    // See https://github.com/flutter/flutter/issues/35293
-    expect(fs.file(fs.path.join(environment.buildDir.path, 'flutter_assets', 'assets/foo/bar.png')).existsSync(), true);
+    expect(fs.file(fs.path.join(environment.buildDir.path, 'flutter_assets', 'assets', 'foo', 'bar.png')).existsSync(), true);
   }));
 
   test('Does not leave stale files in build directory', () => testbed.run(() async {
     await buildSystem.build(const CopyAssets(), environment);
 
-    expect(fs.file(fs.path.join(environment.buildDir.path, 'flutter_assets', 'assets/foo/bar.png')).existsSync(), true);
+    expect(fs.file(fs.path.join(environment.buildDir.path, 'flutter_assets', 'assets', 'foo', 'bar.png')).existsSync(), true);
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart]: Is it important for the test that you get different instances for the two repetitions of:

`fs.file(fs.path.join(environment.buildDir.path, 'flutter_assets', 'assets', 'foo', 'bar.png')` ?

If so, please add a comment. If not, then  stash either the File or the String for the path in a local.

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart]: Fixed

File: packages/flutter_tools/lib/src/build_system/targets/assets.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_system/targets/assets.dart]: Updated bundle.dart

File: packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart]: Fixed

**Review by zanderso — APPROVED** [review]

**Comment by jonahwilliams** [comment]
Fixes https://github.com/flutter/flutter/issues/35293

**Comment by dnfield** [comment]
This also causes https://github.com/flutter/flutter/issues/46163.

I think we want ot keep all these as URI encoded because the framework sends over the file names as URI encoded.  If anything we should make sure we copy _only_ the URI encoded file, and not the unencoded file.

**Comment by jonahwilliams** [comment]
It should be safe to revert this PR. That revert should include tests of the currently failing to load assets, lest we repeat this in 10-12 months

**Pull request merged**